packages/payload/src/admin/fields/Blocks.ts

```diff
@@ -48,24 +48,27 @@ export type BlocksFieldClientComponent = FieldClientComponent<
 export type BlocksFieldLabelServerComponent = FieldLabelServerComponent<
   BlocksField,
   BlocksFieldClientWithoutType
->
+> &
+  FieldPaths
 
 export type BlocksFieldLabelClientComponent =
-  FieldLabelClientComponent<BlocksFieldClientWithoutType>
+  FieldLabelClientComponent<BlocksFieldClientWithoutType> & FieldPaths
 
 type BlockRowLabelBase = {
   blockType: string
   rowLabel: string
   rowNumber: number
 }
 
-export type BlockRowLabelClientComponent = React.ComponentType<
-  BlockRowLabelBase & ClientFieldBase<BlocksFieldClientWithoutType>
->
+export type BlockRowLabelClientProps = BlockRowLabelBase &
+  ClientFieldBase<BlocksFieldClientWithoutType> &
+  FieldPaths
+export type BlockRowLabelClientComponent = React.ComponentType<BlockRowLabelClientProps>
 
-export type BlockRowLabelServerComponent = React.ComponentType<
-  BlockRowLabelBase & ServerFieldBase<BlocksField, BlocksFieldClientWithoutType>
->
+export type BlockRowLabelServerProps = BlockRowLabelBase &
```

Comment:
We send client props to the server component as well so the type needs to include them here. In other places, we'd export a `BlockRowLabelServerPropsOnly` type so we can explicitly type them as needed, like when passed to `RenderServerComponent`. But this way the `BlockRowLabelServerProps` type itself will include all types when building custom components.